### PR TITLE
Ensure extract_latlon_dataarray handles lat/lon as pair

### PR DESCRIPTION
**Pull request by ejgal**
As discussed [here](https://github.com/metno/pyaerocom/pull/126/files#r433793518) extract_latlon_dataarray should check if both the latitude and longitude is contained in the dataarray to ensure that lat and lon have the same length.

### CODE REVIEW

File: pyaerocom/helpers.py [hunk 1]
```diff
@@ -137,21 +137,28 @@ def extract_latlon_dataarray(arr, lat, lon, lat_dimname=None,
...
-        lon = [x for x in lon if lon0 <= x <= lon1]
+        new_lat = []
+        new_lon = []
+        for x, y in zip(lat, lon):
+            if (lat0 <= x <= lat1) and (lon0 <= y <= lon1):
+                new_lat.append(x)
+                new_lon.append(y)
+        if len(new_lat) == 0 and len(new_lon) == 0:
+            raise DataCoverageError('Coordinates not found in dataarray')
+        lat, lon = new_lat, new_lon
```
Comment (ejgal) [inline pyaerocom/helpers.py:148]: A bit more explicit than what we discussed, but I had problems with the list comprehension solution when there were no matches between lat,lon and the coordinates in the input dataarray.

**Comment by ejgal** [comment]
Guess we can hold off on merging this until we figure out the problems in #123.

**Comment by ejgal** [comment]
Changed the naming of the lat/lon ranges.

File: pyaerocom/test/test_colocation.py [hunk 2]
```diff
@@ -8,45 +8,70 @@
...
+
     means = [np.nanmean(coldata.data.data[0]),
              np.nanmean(coldata.data.data[1])]
-    
+
     npt.assert_allclose(means, [obsmean, modmean], rtol=TEST_RTOL)
-    
+
+@testdata_unavail
+def test_colocate_gridded_ungridded_nonglobal(aeronetsunv3lev2_subset):
```
Comment (jgliss) [inline pyaerocom/test/test_colocation.py:50]: Consider simplifying initiation of dummy cube and GriddedData object and also to move iit outside of this method, e.g. into a fixture in conftest, that can then also be used by other tests.

**Comment by jgliss** [comment]
@ejgal Feel free to merge if you consider this done!

File: pyaerocom/test/test_colocation.py (hunk 2 above)
Comment (ejgal) [reply pyaerocom/test/test_colocation.py:50]: Yes, created an issue for this. I would like to think a bit about what would be useful in general before implementing it.
#131.

**Comment by ejgal** [comment]
Used filtering instead of manually selecting stations outside of lat/lon range and merged latest changes in v090dev. Should be ready to go.

**Comment by ejgal** [comment]
@jgliss, merged the whitespace changes, should be ready to go.

**Pull request merged**